### PR TITLE
add disallowed repos policy

### PR DESCRIPTION
**Pull request by sozercan**
**What this PR does / why we need it**:
- adds a disallowed repos example for scenarios such as https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/
- add `generate-all` make target to simplify generation
- docker should run as user to avoid generation of library directory as root (for non-rootless)

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by JaydipGabani** [comment]
If I am not wrong, I think you also need to update [sidebar.js](https://github.com/open-policy-agent/gatekeeper-library/blob/master/website/sidebars.js#L45) and add a path for the policy to show up in website

**Comment by sozercan** [comment]
@JaydipGabani good call. we should automate that https://github.com/open-policy-agent/gatekeeper-library/issues/299

**Review by maxsmythe — APPROVED** [review]
This is intended as the logical inverse of https://github.com/open-policy-agent/gatekeeper-library/tree/master/library/general/allowedrepos ?

LGTM

**Comment by sozercan** [comment]
@maxsmythe that's right

File: src/general/disallowedrepos/src_test.rego [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            "name": "nginx"
+        },
+        "spec": {
+            "initContainers": containers,
+        }
+      }
+     }
+}
+
+input_container_disallowed = [
```
Comment (ritazh) [inline src/general/disallowedrepos/src_test.rego:102]: nit: IMO this will read more clearly if it's `input_container_allowed`

**Pull request merged**